core/state/state_object.go

```diff
@@ -358,37 +377,47 @@ func (s *stateObject) updateTrie(db Database) Trie {
 	if len(s.pendingStorage) > 0 {
 		s.pendingStorage = make(Storage)
 	}
-	return tr
+	return tr, nil
 }
 
-// UpdateRoot sets the trie root to the current root hash of
+// UpdateRoot sets the trie root to the current root hash of. An error
+// will be returned if trie root hash is not computed correctly.
 func (s *stateObject) updateRoot(db Database) {
+	tr, err := s.updateTrie(db)
+	if err != nil {
+		s.setError(fmt.Errorf("updateRoot (%x) error: %v", s.address, err))
```

Comment:
Any particular reason for adding the `updateRoot (addr)` outer context to this error, but not the ones above with TryDelete/TryUpdate?

Reply:
Yeah, in some places, e.g. `updateStateObject` we add a few more context. But I can change it to `s.setError(err)` anyway.

Reply:
changed it to `w`